ts/WoltLabSuite/Core/Core.ts

```diff
@@ -280,16 +278,17 @@ export function enableLegacyInheritance<T>(legacyClass: T): void {
 
     const constructed = Reflect.construct(legacyClass as any, args, thisValue.constructor);
     Object.entries(constructed).forEach(([key, value]) => {
+      if (typeof value === "function") {
+        value = value.bind(thisValue);
+      }
+
       thisValue[key] = value;
     });
 
-    let object = thisValue;
-    while ((object = Object.getPrototypeOf(object))) {
-      Object.getOwnPropertyNames(object).forEach((name) => {
-        if (typeof object[name] === "function" && !defaultFunctions.includes(name)) {
-          object[name] = object[name].bind(thisValue);
-        }
-      });
+    for (const key in thisValue) {
+      if (typeof thisValue[key] === "function") {
+        constructed[key] = thisValue[key].bind(thisValue);
+      }
```

Comment:
Unfortunately, it is. The issue is caused by the event binding in the parent constructor, it will invoke the callback with `this === constructed`. Forcefully binding the method to `thisValue` will still call `constructed.thisIsMyCallback`, but with `this === thisValue`.

I do not pretend to fully understand how this works, because we're messing with the prototype twice. Once in `Core.inherit()` and then on runtime with the `call()` to the super constructor. Unfortunately, any return value of `Core.inherit()` is discarded, therefore we can't make use of `Proxy` to safely "extend" an ES6 class.

So far I have not observed any drawbacks from my approach, all test cases passed just fine. I'm not even going to pretend that it is pretty ;)